src/sentry/middleware/auth.py

```diff
@@ -80,3 +102,43 @@ def process_exception(self, request: Request, exception):
             from sentry.web.frontend.accounts import expired
 
             return expired(request, exception.user)
+
+
+class HybridCloudAuthenticationMiddleware(MiddlewareMixin):
+    def process_request(self, request: Request):
+        from sentry.web.frontend.accounts import expired
+
+        auth_result = auth_service.authenticate(request=authentication_request_from(request))
+        request.user_from_signed_request = auth_result.user_from_signed_request
```

Comment:
Longer term it would be nice to go back to one AuthenticationMiddleware and have this service handle the different logic required for monolith/silo mode.

Reply:
In the future, Monolith mode implementation becomes deprecated and we basically only use HybridCloudAuthenticationMiddleware.  But this affords us safety by ensuring that this change *cannot impact the existing saas behavior*.  I felt authentication was the one case where splitting the implementation until we can fully vet this was the correct move.

Reply:
That's fair. Reducing risk in the present is a good goal to have.